### PR TITLE
Add Asset DTO

### PR DESCRIPTION
**Pull request by stark-dev**
Signed-off-by: Mauro Guerrera <mauroguerrera@eaton.com>

### CODE REVIEW

File: asset/CMakeLists.txt [hunk 1]
```diff
@@ -32,8 +34,13 @@ etn_target(shared ${PROJECT_NAME}-libng
         src/asset-helpers.cpp
         src/asset-db.cpp
         src/asset-licensing.cpp
+        src/asset-notifications.cpp
         src/asset-import.cpp
         src/asset-configure-inform.cpp
+        src/messagebus/message-bus.h
+        src/messagebus/message-bus.cpp
+        src/messagebus/message.h
+        src/messagebus/message.cpp
```
Comment (etnKlendathu) [inline asset/CMakeLists.txt:43]: 6th copy.... but fty-common-messagebus has pull request on this :(

File: asset/src/asset-db.cpp [hunk 2]
```diff
@@ -13,6 +14,83 @@ namespace fty::asset::db {
...
+    )";
+    return sql;
+}
+
+static void fetchDto(const fty::db::Row& row, Dto& asset)
+{
+    uint32_t assetId;
+
+    std::string tmp;
+    row.get("id", assetId);
```
Comment (etnKlendathu) [inline asset/src/asset-db.cpp:50]: assetId = row.get<uint32_t>("id");

File: asset/src/asset-db.cpp [hunk 3]
```diff
@@ -13,6 +14,83 @@ namespace fty::asset::db {
...
+    row.get("internalName", tmp);
+    asset.name = tmp;
```
Comment (etnKlendathu) [inline asset/src/asset-db.cpp]: asset.name = row.get("internalName");

File: asset/src/asset-db.cpp [hunk 4]
```diff
@@ -13,6 +14,83 @@ namespace fty::asset::db {
...
+    row.get("parent", tmp);
+    asset.parent = tmp;
+    row.get("status", tmp);
+    if (tmp == "active") {
+        asset.status = Dto::Status::Active;
+    } else if (tmp == "nonactive") {
+        asset.status = Dto::Status::Nonactive;
+    } else {
+        asset.status = Dto::Status::Unknown;
+    }
```
Comment (etnKlendathu) [inline asset/src/asset-db.cpp]: maybe better somewhere in the top of function getStatus =[](const std::string& st) {
   if (tmp == "active") {
       return Dto::Status::Active;
   }
   ....
}; 

and inplace
asset.status = getStatus(row.get("status"));

File: asset/src/asset-db.cpp [hunk 5]
```diff
@@ -13,6 +14,83 @@ namespace fty::asset::db {
...
+    int32_t priority;
+    row.get("priority", priority);
+    asset.priority = priority;
+
+    auto attributes = selectExtAttributes(assetId);
+    if (!attributes) {
+        throw std::runtime_error(fmt::format("Failed to get external attributes for asset {}", assetId));
+    }
+
+    for (auto el : *attributes) {
```
Comment (etnKlendathu) [inline asset/src/asset-db.cpp:76]: const auto& el

or better 
const auto& [key, value]:

File: asset/src/asset-db.cpp [hunk 6]
```diff
@@ -1498,6 +1600,38 @@ Expected<std::vector<uint32_t>> selectAssetDeviceLinksSrc(uint32_t elementId)
...
+    )";
+
+    try {
+        fty::db::Connection    conn;
+        std::vector<LinkEntry> links;
+        for (const auto& it : conn.select(sql, "asset_id"_p = elementId)) {
+            LinkEntry l;
+            l.source    = it.get<std::string>("name");
+            l.link_type = it.get<int32_t>("linkType");
+            l.src_out   = it.get<std::string>("srcOut");
```
Comment (etnKlendathu) [inline asset/src/asset-db.cpp:1629]: you could just use it.get("...") for string template is not needed

**Review by etnKlendathu — APPROVED** [review]
good job. I hope it works as a charm... :)

File: asset/src/asset-db.cpp [hunk 7]
```diff
@@ -13,6 +16,84 @@ namespace fty::asset::db {
...
+
+static void fetchDto(const fty::db::Row& row, Dto& asset)
+{
+    uint32_t assetId;
+
+    std::string tmpStr;
+    int32_t     tmpInt;
+    row.get("id", assetId);
+    row.get("internalName", tmpStr);
+    asset.name = tmpStr;
```
Comment (etnKlendathu) [inline asset/src/asset-db.cpp:52]: asset.name = row.get("internalName");

File: asset/src/asset-db.cpp (hunk 5 above)
Comment (stark-dev) [reply asset/src/asset-db.cpp:76]: since when?? <3 I love it!

**Pull request merged**